File: java/nim_array.java
import java.util.Scanner;

public class nim_array {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        System.out.print("Masukkan NIM : ");
        String nim = sc.nextLine();

        int[] digit = new int[nim.length()];
        for (int i = 0; i < nim.length(); i++) {
            digit[i] = Character.getNumericValue(nim.charAt(i));
        }

        System.out.print("Digit        : [ ");
        for (int i = 0; i < digit.length; i++) {
            System.out.print(digit[i]);
            if (i < digit.length - 1) System.out.print(", ");
        }
        System.out.println(" ]");

        int total = 0;
        for (int d : digit) total += d;

        int maks = digit[0];
        for (int d : digit)
            if (d > maks) maks = d;

        int minimum = digit[0];
        for (int d : digit)
            if (d < minimum) minimum = d;

        double rata = (double) total / digit.length;

        int[] rev = new int[digit.length];
        for (int i = 0; i < digit.length; i++) {
            rev[i] = digit[digit.length - 1 - i];
        }

        System.out.println("Total        : " + total);
        System.out.println("Maksimum     : " + maks);
        System.out.println("Minimum      : " + minimum);
        System.out.println("Rata-rata    : " + rata);

        System.out.print("Reverse      : [ ");
        for (int i = 0; i < rev.length; i++) {
            System.out.print(rev[i]);
            if (i < rev.length - 1) System.out.print(", ");
        }
        System.out.println(" ]");

        sc.close();
    }
}

File: php/nim_array.php
<?php

$nim = readline("Masukkan NIM: ");

$digit = [];
for ($i = 0; $i < strlen($nim); $i++) {
    $digit[$i] = intval($nim[$i]);
}

    $total = 0;
    for ($i = 0; $i < count($digit); $i++) {
        $total += $digit[$i];
}

    $maks = $digit[0];
    for ($i = 0; $i < count($digit); $i++) {
        if ($digit[$i] > $maks) {
            $maks = $digit[$i];
        }
    }

    $minim = $digit[0];
    for ($i = 0; $i < count($digit); $i++){
        if ($digit[$i] < $minim) {
            $minim = $digit[$i];
        }
    }

    $rata = $total / count($digit);

    $rev = [];
    for ($i = count($digit) - 1; $i >= 0; $i--) {
        $rev[] = $digit[$i];
    }

echo "Digit        : [ " . implode(", ", $digit) . " ]\n";
echo "Total        : $total\n";
echo "Maksimum     : $maks\n";
echo "Minimum      : $minim\n";
echo "Rata-rata    : $rata\n";
echo "Reverse array: [ " . implode(", ", $rev) . " ]\n";

?>